### PR TITLE
Rename to rollmint

### PR DESCRIPTION
**Pull request by tzdybal**
- [x] rename go module
- [x] rename imports
- [x] rename all comments
- [x] rename variables/structs/etc with opti and optimint in their name
- [x] fix links (badges/etc)
- [x] update ADRs
- [x] grep entire directory to ensure full rename

It's a part of: #535

### CODE REVIEW

**Comment by tzdybal** [comment]
@jeremysklaroff added you because of: 2de7d621cc3550a53148704a9caa6576d6a52a45

File: .github/workflows/docker-build-ethermint-debug.yml [hunk 1]
```diff
@@ -40,10 +40,10 @@ jobs:
           ls
           mv ethermint ..
           cd ..
-          cp -R optimint ethermint
+          cp -R rollmint ethermint
```
Comment (MSevey) [inline .github/workflows/docker-build-ethermint-debug.yml:43]: nit: we are renaming it once which means we might rename it again. Or if someone forks this and wants to name it differently they are going to need to find and replace again. 

So could we just use a `env` variable up top? Like `DIR_NAME`?

You could then use that var in a number of places.

File: proto/get_deps.sh [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 cd "$(dirname "${BASH_SOURCE[0]}")"
 
-TM_VERSION=v0.34.14
+TM_VERSION=v0.34.21
```
Comment (MSevey) [inline proto/get_deps.sh:5]: sneaky

File: proto/tendermint/abci/types.proto [hunk 3]
```diff
@@ -212,6 +211,12 @@ message ResponseCheckTx {
   repeated Event events     = 7
       [(gogoproto.nullable) = false, (gogoproto.jsontag) = "events,omitempty"];
   string codespace = 8;
+  string sender    = 9;
```
Comment (MSevey) [inline proto/tendermint/abci/types.proto:214]: This doesn't feel like a rename :-) 

Is this bringing in some other changes to make the merge easier?

File: .github/workflows/docker-build-test-ethermint.yml [hunk 4]
```diff
@@ -37,16 +37,16 @@ jobs:
         with:
           repository: celestiaorg/ethermint
           path: ethermint
-          ref: optimint-v0.3.0-rebase
+          ref: rollmint-v0.3.0-rebase
```
Comment (MSevey) [inline .github/workflows/docker-build-test-ethermint.yml:40]: nit: Can't you use `TAG_PREFIX` here?

File: .github/workflows/docker-build-test-ethermint.yml [hunk 5]
```diff
@@ -91,14 +91,14 @@ jobs:
       with:
         repository: celestiaorg/ethermint
         path: ethermint
-        ref: optimint-v0.3.0-rebase
+        ref: rollmint-v0.3.0-rebase
```
Comment (MSevey) [inline .github/workflows/docker-build-test-ethermint.yml:94]: same `TAG_PREFIX`

File: .github/workflows/docker-build-test-ethermint.yml [hunk 6]
```diff
@@ -91,14 +91,14 @@ jobs:
...
     - name: Set up Go
       uses: actions/setup-go@v2
       with:
         go-version: 1.17
     - name: "Setup Cluster"
       run: |
-        export ETHERMINT_IMAGE_TAG=optimint-$(git rev-parse --short "$GITHUB_SHA")
+        export ETHERMINT_IMAGE_TAG=rollmint-$(git rev-parse --short "$GITHUB_SHA")
```
Comment (MSevey) [inline .github/workflows/docker-build-test-ethermint.yml:101]: same `TAG_PREFIX`

**Review by MSevey — APPROVED** [review]
All my comments are nits really and could be handled in a small follow up if necessary.

File: proto/tendermint/abci/types.proto (hunk 3 above)
Comment (tzdybal) [reply proto/tendermint/abci/types.proto:214]: This is the result of updating TM to 0.34.21. I'll create separate PR for this; I missed this while working on #528.

File: .github/workflows/docker-build-ethermint-debug.yml (hunk 1 above)
Comment (tzdybal) [reply .github/workflows/docker-build-ethermint-debug.yml:43]: Created #546,

File: proto/get_deps.sh (hunk 2 above)
Comment (tzdybal) [reply proto/get_deps.sh]: My bad. Reverted this change and created: https://github.com/celestiaorg/optimint/pull/547

**Review by MSevey — APPROVED** [review]
Docker Github Actions are failing but otherwise looks good

**Comment by tzdybal** [comment]
Those jobs are expected to fail, sorry for not mentioning this. I need to update cosmos-sdk to use proper version of rollmint, to "fix" those builds. (But I need "rollmint" in the first place)

**Review by MSevey — APPROVED** [review]

**Pull request merged**